drivers/sensor/sensor_shell.c

```diff
@@ -349,21 +443,32 @@ static int cmd_get_sensor(const struct shell *sh, size_t argc, char *argv[])
 	iodev_sensor_shell_read_config.sensor = dev;
 	iodev_sensor_shell_read_config.count = count;
 
-	struct sensor_shell_processing_context ctx = {
-		.dev = dev,
-		.sh = sh,
-	};
+	ctx.dev = dev;
+	ctx.sh = sh;
 	err = sensor_read(&iodev_sensor_shell_read, &sensor_read_rtio, &ctx);
 	if (err < 0) {
 		shell_error(sh, "Failed to read sensor: %d", err);
 	}
-	sensor_processing_with_callback(&sensor_read_rtio, sensor_shell_processing_callback);
 
 	k_mutex_unlock(&cmd_get_mutex);
 
 	return 0;
 }
 
+static void sensor_shell_processing_entry_point(void *a, void *b, void *c)
+{
+	ARG_UNUSED(a);
+	ARG_UNUSED(b);
+	ARG_UNUSED(c);
+
+	while (true) {
+		sensor_processing_with_callback(&sensor_read_rtio,
+						sensor_shell_processing_callback);
+	}
+}
+K_THREAD_DEFINE(sensor_shell_processing_tid, CONFIG_SENSOR_SHELL_THREAD_STACK_SIZE,
+		sensor_shell_processing_entry_point, NULL, NULL, NULL, 0, 0, 0);
```

Comment:
I know this is useful for testing the new streaming feature, but there are other users of the sensor shell that don't want to take the memory hit. Can you please partition the sensor shell changes so the it can still be used with `CONFIG_SENSOR_ASYNC_API=n`?

Reply:
Done